sdk/helper/keysutil/policy.go

```diff
@@ -1385,7 +1385,17 @@ func (p *Policy) Import(ctx context.Context, storage logical.Storage, key []byte
 	} else {
 		parsedPrivateKey, err := x509.ParsePKCS8PrivateKey(key)
 		if err != nil {
-			return fmt.Errorf("error parsing asymmetric key: %s", err)
+			if strings.Contains(err.Error(), "unknown elliptic curve") {
```

Comment:
This seems a little brittle against possible future changes from an error message? 

Are there any reasons we don't just try to call `ParsePKCS8Ed25519PrivateKey` on any err?

Reply:
We might have a bad ASN.1 blob or something else, that'd trigger a non-unknown curve error. If that's the case, the Ed25519 parsing won't do anything, but give a weird, unrelated error message.